src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingChannelExistenceChecker.java

```diff
@@ -0,0 +1,293 @@
+
+        // Handle the exceptional status code
+        HttpResponse httpResponse =
+            handleExceptionalStatus(
+                closeableHttpResponse,
+                null,
+                ServiceResponseHandler.ApiName.STREAMING_CHANNEL_STATUS,
+                httpClient,
+                request,
+                requestBuilder);
+
+        // Grab the string version of the response entity
+        String blob = consumeAndReturnResponseEntityAsString(httpResponse.getEntity());
+
+        // Read out our blob into a pojo
+        channelExistenceCheckerResponse =
+            objectMapper.readValue(blob, ChannelExistenceCheckerResponse.class);
+      }
+
+      if (channelExistenceCheckerResponse.getStatusCode() == SUCCESS.getStatusCode()) {
+        assert channelExistenceCheckerResponse.getChannels().size() == 1;
+        if (channelExistenceCheckerResponse
+            .getChannels()
+            .get(0)
+            .getStatusCode()
+            .equals(ERR_CHANNEL_DOES_NOT_EXIST_OR_IS_NOT_AUTHORIZED.getStatusCode())) {
+          LOGGER.debug(String.format("%s Channel:%s doesn't exist", DEBUG_PREFIX, channelName));
+          return false;
+        } else if (channelExistenceCheckerResponse
+                .getChannels()
+                .get(0)
+                .getStatusCode()
+                .equals(ERR_CHANNEL_HAS_INVALID_CLIENT_SEQUENCER.getStatusCode())
+            || channelExistenceCheckerResponse
+                .getChannels()
+                .get(0)
+                .getStatusCode()
+                .equals(SUCCESS.getStatusCode())) {
+          // We check two status codes, one for invalid client sequencers and one for a valid client
+          // Sequencer
+          LOGGER.debug(String.format("%s Channel:%s does exist", DEBUG_PREFIX, channelName));
+          return true;
+        }
+      }
+    } catch (Exception e) {
+      LOGGER.error(
+          "{} Error Getting a valid response for ChannelName:{} error:{}",
+          DEBUG_PREFIX,
+          channelName,
+          e.getMessage());
+    }
+    return false;
+  }
+
+  /** This implementation is copied over from Ingest SDK. */
+  private static HttpResponse handleExceptionalStatus(
+      HttpResponse response,
+      UUID requestId,
+      ServiceResponseHandler.ApiName apiName,
+      CloseableHttpClient httpClient,
+      HttpUriRequest request,
+      RequestBuilder requestBuilder)
+      throws IOException, BackOffException {
+    if (!isStatusOK(response.getStatusLine())) {
+      StatusLine statusLine = response.getStatusLine();
+      LOGGER.warn(
+          "{} Status hit from {}, requestId:{}",
+          statusLine.getStatusCode(),
+          apiName,
+          requestId == null ? "" : requestId.toString());
+
+      // if we have a 503 exception throw a backoff
+      switch (statusLine.getStatusCode()) {
+          // If we have a 503, BACKOFF
+        case HttpStatus.SC_SERVICE_UNAVAILABLE:
+          throw new BackOffException();
```

Comment:
Do we want to retry for this BackOffException? We don't retry in checkChannelExistance

Reply:
good catch, I copied this function from sdk and I do see we dont have retries in channelexistence, I plan to add retries, but let me check this..